src/message.rs

```diff
@@ -1,19 +1,44 @@
-use serde::{Deserialize, Serialize};
+use std::collections::HashMap;
 use std::fmt;
 
+use chrono::{DateTime, Utc};
+use serde::{Deserialize, Serialize};
+
 #[derive(Serialize, Deserialize, Debug)]
 pub struct Message {
-    payload: String,
+    timestamp: DateTime<Utc>,
+    source: String,
+    payload: HashMap<String, String>,
 }
 
 impl Message {
-    pub fn new(payload: String) -> Message {
-        Message { payload }
+    pub fn new(source: &str) -> Message {
+        Message {
+            timestamp: Utc::now(),
+            source: String::from(source),
+            payload: HashMap::new(),
+        }
+    }
+
+    pub fn insert_string(&mut self, field: &str, value: &str) {
```

Comment:
This function name could be better, like `insert_data` or `insert_payload_item`or something like that.

Reply:
I was first thinking that I would implement functions for adding different types. For example, `insert_string` for strings and `insert_i32` for integers.

Today I got a better idea when I saw your comment! I realized I can use `Into` trait to allow the caller to give any type that supports type conversion to `PayloadValue` enum. The code is a bit verbose with all the different `From` trait implementations but it works now very nicely.